m3/src/main/java/com/uber/m3/tally/m3/M3Reporter.java

```diff
@@ -379,6 +379,10 @@ public void reportHistogramValueSamples(
             // We know that the HashMap will only contain two items at this point,
             // therefore initialCapacity of 2 and loadFactor of 1.
             tags = new HashMap<>(2, 1);
+        } else {
+            // Copy over the map since it might be unmodifiable and, even if it's
+            // not, we don't want to modify it.
+            tags = new HashMap<>(tags);
```

Comment:
was this revealed in a test? or prod? let's make sure we add tests either way

Reply:
This does not belong to this PR. It was added as part of https://github.com/uber-java/tally/pull/38